benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/scheduler/ExperimentTaskScheduler.java

```diff
@@ -146,42 +156,37 @@ public synchronized void handleRunningExperiment(String experimentID) {
 
       experimentState = experimentModelDAO.getExperimentState(experimentID);
       runningState = experimentModelDAO.getRunningState(experimentID);
+      logger.info("handleRunningExperiment: experimentState == " + experimentState);
+      logger.info("handleRunningExperiment: runningState == " + runningState);
 
-      // Stop when we reach a final state
-      while (!exit) {
+      while (true) {
 
         prevExperimentState = experimentState;
         prevRunningState = runningState;
 
         try {
-          experimentState = handleExperimentState(experimentID);
+          handleExperimentState(experimentID);
         } catch (BenchFlowExperimentIDDoesNotExistException e) {
           e.printStackTrace();
           break;
         }
 
+        experimentState = experimentModelDAO.getExperimentState(experimentID);
         runningState = experimentModelDAO.getRunningState(experimentID);
-        logger.info("handleRunningExperiment: runningState == " + runningState);
-        logger.info("handleRunningExperiment: prevRunningState == " + prevRunningState);
+        logger.info("handleRunningExperiment: (while) experimentState == " + experimentState);
+        logger.info("handleRunningExperiment: (while) runningState == " + runningState);
+        logger.info("handleRunningExperiment: (while) prevRunningState == " + prevRunningState);
 
-        // Exit as soon as final state is executed
+        // Exit as soon as the final state has been executed
         if (prevExperimentState == TERMINATED) {
-          exit = true;
+          break;
         }
 
-        /*
-        Exit while waiting for the Faban Manager to notify about the result of the scheduled
-        trial. This exits before the execution of HANDLE_EXPERIMENT_RESULT
-        
-        NOTE for TERMINATING: Check depends if the system is waiting for input from the FabanManager
-        or not. So in the inner loop for running experiment we exit, but in the outer loop (HERE) we
-        check if we expect a result from the FabanManager or not. If we do not expect it we run the
-        loop again so that the TERMINATED state handling gets executed.
-        */
-        else if (runningState == RunningState.HANDLE_TRIAL_RESULT
-            || runningState == RunningState.TERMINATING
-                && prevRunningState == RunningState.HANDLE_TRIAL_RESULT) {
-          exit = true;
+
+        // Exit while waiting for the Faban Manager to notify about the result of the scheduled
+        // trial. This exits before the execution of HANDLE_EXPERIMENT_RESULT
+        if (runningState == RunningState.HANDLE_TRIAL_RESULT) {
```

Comment:
it is not necessary anymore because we do the check when aborting https://github.com/benchflow/benchflow/pull/534/files/e07325058d8fb69b69981f8bf3470b6eb9f1c989..4723add973de6e9386fa66562e5649a44b3a640e#diff-e320626f292c5fd5e3bf5ea062ab6c90R383. 
The TERMINATING -> TERMINATED is then the same in all cases and it terminates once the state TERMINATED has been executed (line 181 above)